### PR TITLE
Report more detailed unwinding error

### PR DESCRIPTION
**Pull request by florian-kuebler**
In case of an unwinding error during leaf function patching,
where we already know its a leaf function, we should ensure
to tell the user if the stack dump was too small (as this is
actionable).

Test: Inspect unwind errors on sample game.

### CODE REVIEW

File: src/LinuxTracing/LeafFunctionCallManager.cpp [hunk 1]
```diff
@@ -130,6 +130,11 @@ Callstack::CallstackType LeafFunctionCallManager::PatchCallerOfLeafFunction(
   // If the caller is not executable, we have an unwinding error.
   unwindstack::MapInfo* caller_map_info = current_maps->Find(libunwindstack_leaf_caller_pc);
   if (caller_map_info == nullptr || (caller_map_info->flags() & PROT_EXEC) == 0) {
+    // If the error was because the stack sample was too small, the user can act and increase the
+    // stack size. So we report that case separately.
+    if (stack_size > stack_dump_size_) {
```
Comment (dpallotti) [inline src/LinuxTracing/LeafFunctionCallManager.cpp:135]: Soo, I need to understand before approving: the stack dump is too small, but libunwindstack tries to unwind anyway, reports the new frame, but makes a mistake that we detect looking at the `MapInfo`? Is that the case?

File: src/LinuxTracing/LeafFunctionCallManager.cpp [hunk 2]
```diff
@@ -130,6 +130,11 @@ Callstack::CallstackType LeafFunctionCallManager::PatchCallerOfLeafFunction(
   // If the caller is not executable, we have an unwinding error.
   unwindstack::MapInfo* caller_map_info = current_maps->Find(libunwindstack_leaf_caller_pc);
   if (caller_map_info == nullptr || (caller_map_info->flags() & PROT_EXEC) == 0) {
+    // If the error was because the stack sample was too small, the user can act and increase the
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/LinuxTracing/LeafFunctionCallManager.cpp]: Done.

File: src/LinuxTracing/LeafFunctionCallManager.cpp (hunk 1 above)
Comment (florian-kuebler) [reply src/LinuxTracing/LeafFunctionCallManager.cpp:135]: Yes, exactly this is happening.

Comment (florian-kuebler) [reply src/LinuxTracing/LeafFunctionCallManager.cpp:135]: The point is, that assuming $rbp is untouched ("stack_size" would be two complete frames), we might be still completely fine when unwinding the first frame even if "stack_size" is larger than the "stack dump size".

Comment (dpallotti) [reply src/LinuxTracing/LeafFunctionCallManager.cpp:135]: Ok, so this is a case of speculative frame that might or might not be correct?

**Review by dpallotti — APPROVED** [review]

File: src/LinuxTracing/LeafFunctionCallManager.cpp (hunk 1 above)
Comment (florian-kuebler) [reply src/LinuxTracing/LeafFunctionCallManager.cpp:135]: Yes, I think so.

**Pull request merged**